From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sat, 29 Jul 2023 17:33:24 +0200
Subject: rt5640: Keep the codec enabled when idle

This makes analog audio work even without any i2s streams being enabled.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 sound/soc/codecs/rt5640.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/codecs/rt5640.c b/sound/soc/codecs/rt5640.c
index b27a31de9633..e7e812263adf 100644
--- a/sound/soc/codecs/rt5640.c
+++ b/sound/soc/codecs/rt5640.c
@@ -2957,6 +2957,8 @@ static const struct snd_soc_component_driver soc_component_dev_rt5640 = {
 	.num_dapm_widgets	= ARRAY_SIZE(rt5640_dapm_widgets),
 	.dapm_routes		= rt5640_dapm_routes,
 	.num_dapm_routes	= ARRAY_SIZE(rt5640_dapm_routes),
+	.suspend_bias_off	= 1,
+	.idle_bias_on		= 1,
 	.use_pmdown_time	= 1,
 	.endianness		= 1,
 };
-- 
Armbian

